src/main/java/com/upserve/uppend/metrics/AppendOnlyStoreWithMetrics.java

```diff
@@ -276,6 +266,47 @@ public void close() throws Exception {
             store.close();
         } finally {
             context.stop();
+            removeGaugeMetrics();
+        }
+    }
+
+    // Returns a unique key for this instance of the nth gague
+    private String getKeyForGauge(int n) {
+        return MetricRegistry.name(rootName, UPPEND_APPEND_STORE, store.getName(), GAUGE_NAMES[n], String.valueOf(store.hashCode()));
```

Comment:
Why would the store name not be unique enough?

Reply:
Some `Stores` are opened twice, pointing to different paths, once for the kinesis process and once for the formulator process. That name collision is the cause of the crashes in Gastronomer when using this version of `Uppend`.

If using the `hashCode()` is problematic, we could fix this on the Gastronomer side by passing in a `rootName` that includes more context or adding allowing another name value in the constructor...

Reply:
I think that would be better for the metrics, as long as we aren't accidentally tying the metric name to the store name and can pass that context easily.